### ISSUE TITLE
Enable user to give feedback to the demo owner.

### ISSUE THREAD

**Issue by uttu357** [Cloud-CV/Origami#198]
- On the demo page, the user should be able to give feedback to the demo owner.
- The demo owner should be able to see the feedback for each of his demos.

**Comment by vpn1997** [comment]
Hey rather than one to one feedback , I think we should provide something like comment section where demo creator as well as other users can give their reviews ,ask their doubts etc. :smile:

**Comment by uttu357** [comment]
@deshraj What do you think about this?

**Comment by vpn1997** [comment]
@uttu357 hey can i work on this ? 
Clicking on comment button will take us to comment section.

![screenshot 4](https://user-images.githubusercontent.com/17298412/38517368-687d180c-3c57-11e8-933d-ab9002d83886.png)

- I have made the initial attempt for demo section 
( https://drive.google.com/file/d/190cGlDfKWq8jREIjT_XPCeCD5gG-XQnQ/view )

**Comment by fristonio** [comment]
My thoughts on this are :  

* I would like to go with the idea of threads like comments rather than one to one feedback.
* For the implementation, I think it would be better to have a sidebar having the comment threads, this sidebar can have hidden and visible state. S rather than a comment button we will have a comment icon on top right corner which on clicking will open the sliding sidebar on the right, making everything else out of focus.
* Starring the demo should come under origami-insights section and I am currently working on it.
* There should not be upvote downvote or stars feature for comment. We should keep it as simple as it can be with the sole purpose of feedback.

**Comment by deshraj** [comment]
> I would like to go with the idea of threads like comments rather than one to one feedback.

@fristonio I agree with your point that we should keep the comments in threads.

> For the implementation, I think it would be better to have a sidebar having the comment threads, this sidebar can have hidden and visible state. S rather than a comment button we will have a comment icon on top right corner which on clicking will open the sliding sidebar on the right, making everything else out of focus.

Instead of putting the comments in the sidebar, I think we should put the comments section at the bottom on the demo instead of putting it in the sidebar. Not everyone wants to see the comments related to a demo. People are more interested in seeing the demo. We can utilize the sidebar for something else like showing other details/analytics or something else. 

> Starring the demo should come under origami-insights section and I am currently working on it.

Starring the demo can come on the main demo page itself on the top similar to github.

> There should not be upvote downvote or stars feature for comment. We should keep it as simple as it can be with the sole purpose of feedback.

Yeah, I agree. Lets keep it simple.

**Comment by fristonio** [comment]
> Instead of putting the comments in the sidebar, I think we should put the comments section at the bottom on the demo instead of putting it in the sidebar. Not everyone wants to see the comments related to a demo. People are more interested in seeing the demo. We can utilize the sidebar for something else like showing other details/analytics or something else.

I was thinking that a separate view like a portal or something would be better suited for analytics.
For the comment section, since the two sections(demo and comments) are better off separate I thought that we can have a different section entirely for comments which do not interfere with the flow of the demo. 
Though the flow you described also seems like a good approach and we can work on those lines too. :)

**Comment by uttu357** [comment]
@vpn1997 @fristonio I would like you to complete the work you are doing currently before taking up another issue. Plus adding new features is not the priority. Fixing UI bugs and making UI stable is. It would be preferable if that is completed first so that we have a stable Origami application before we start adding new features.

**Comment by deshraj** [comment]
I totally agree with @uttu357 that we should try to complete the existing moving parts and get them done and make the project more stable before start implementing new features. Let's make sure that we are more focused on improving the current project instead of adding new features.

**Comment by deshraj** [comment]
@vpn1997 the features that you are suggesting are really good but they are overkill at the moment. We should first try to have basic complete functionalities and then, later on, we can try to add more things on top of it.

**Comment by Isha1233** [comment]
I'm thinking to implement :-
Feedback Form:
  - Fields: User Name (optional), Rating (1-5 stars), Comments, Suggestions.
  
 Feedback Submission:
  - Store feedback in a database linked to the demo (Demo ID, User ID, Timestamp).

Dashboard for Demo Owners:
  - Display:
    - List of demos with feedback counts.
    - Average ratings.
    - Detailed comments and suggestions.

Optional Features:
  - Notifications for new feedback.
  - Moderation for inappropriate comments.
